dspace-api/src/main/java/org/dspace/content/RelationshipServiceImpl.java

```diff
@@ -94,18 +98,32 @@ public Relationship create(Context context, Relationship relationship) throws SQ
     }
 
     @Override
-    public void updatePlaceInRelationship(Context context, Relationship relationship, boolean isCreation)
+    public void updatePlaceInRelationship(Context context, Relationship relationship)
         throws SQLException, AuthorizeException {
         Item leftItem = relationship.getLeftItem();
+        // Max value is used to ensure that these will get added to the back of the list and thus receive the highest
+        // (last) place as it's set to a -1 for creation
+        if (relationship.getLeftPlace() == -1) {
+            relationship.setLeftPlace(Integer.MAX_VALUE);
+        }
+        Item rightItem = relationship.getRightItem();
+        if (relationship.getRightPlace() == -1) {
+            relationship.setRightPlace(Integer.MAX_VALUE);
```

Comment:
This method's logic  is increasingly confusing.  Can you better explain why we have to temporarily set the place to `MAX_VALUE`?  Why not just set it to the length of `leftRelationships` or `rightRelationships` on the next lines?

Reply:
The place -1 implies it's not set and should be appended at the end.
The length of the arrays is also not necessarily the correct place, e.g. if there are plain text authors present

A comment has been included at https://github.com/DSpace/DSpace/pull/2551/files/e56e71a90338000666a2ce012c3849fc56d0c639#diff-2e89fe18482e7ac43a1a4e901a8105b7R104 to explain this